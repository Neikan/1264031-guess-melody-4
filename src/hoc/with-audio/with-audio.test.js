import React from "react";
import PropTypes from "prop-types";
import renderer from "react-test-renderer";
import withAudio from "./with-audio";


const MockComponent = (props) => {
  const {children} = props;

  return (
    <div>
      {children}
    </div>
  );
};

MockComponent.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired,
};

const MockComponentWrapped = withAudio(MockComponent);

describe(`Test withAudio component`, () => {
  test(`withAudio is rendered correctly`, () => {
    const tree = renderer.create((
      <MockComponentWrapped
        id={`id-1`}
        src={`src-1`}
        isPlaying={false}
      />
    ), {
      createNodeMock() {
        return {};
      }
    }).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
