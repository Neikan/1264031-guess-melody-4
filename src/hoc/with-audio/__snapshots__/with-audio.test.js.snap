// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test withAudio component withAudio is rendered correctly 1`] = `
<div>
  <audio />
</div>
`;
